refactor: virt-handler

sarthaksarthak9 opened the pull request:
### What this PR does
After this PR:
- Replaced `api.NewMinimalVMI` and `api.NewMinimalVMIWithNS` with `libvmi.New`.
- Used `libvmi.WithName` and `libvmi.WithNamespace` for setting VMI properties.
- Maintained existing functionality and comments without adding new explanations.

Fixes #14149

### Why we need it and why it was done in this way
It will replaces all occurrences of `api.NewMinimalVMI` and `api.NewMinimalVMIWithNS` with `libvmi.New` to promote the usage of the `libvmi` package for creating VirtualMachineInstance objects.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
none
```

0xFelix commented on pkg/virt-handler/cmd-client/client_test.go:
```diff
@@ -52,7 +51,9 @@ var _ = Describe("Virt remote commands", func() {
+		vmi = libvmi.New(
+			libvmi.WithName("testvmi"),
+		)
 		vmi.UID = "1234"
```
There could be options in `libvmi` to set these fields too. Please check.

0xFelix commented on pkg/virt-handler/container-disk/mount_test.go:
```diff
@@ -54,7 +53,9 @@ var _ = Describe("ContainerDisk", func() {
 		tmpDir, err = os.MkdirTemp("", "containerdisktest")
 		Expect(err).ToNot(HaveOccurred())
-		vmi = api.NewMinimalVMI("fake-vmi")
+		vmi = libvmi.New(
+			libvmi.WithName("fake-vmi"),
+		)
 		vmi.UID = "1234"
```
Same. If there is not already an option for `UID` it might make sense to add one.

sarthaksarthak9 commented:
@0xFelix Thank you for your thorough review of the PR. I’ve addressed all your comments in this revision. Please feel free to let me know if there’s anything else to refine—appreciating your insights!

0xFelix commented on pkg/virt-handler/cmd-client/client_test.go:
```diff
@@ -52,8 +51,10 @@ var _ = Describe("Virt remote commands", func() {
+			libvmi.WithName("testvmi"),
+			libvmi.WithUID("1234"),
+		)
 		vmi.Status = v1.VirtualMachineInstanceStatus{
```
There is a `libvmi` option called `status.New` and there is an option `status.WithActivePod` for it. Please update the PR to make use of them.

sarthaksarthak9 replied:
The function `status.WithActivePod` returns a `status.Option` type and the `libvmi.New` function expects a `libvmi.Option` type. I attempted to resolve this by using type conversion. However, this didn’t work, and I’m still getting the same error. Should status.WithActivePod be refactored to return a libvmi.Option type?

0xFelix replied:
It is used like this:

```
vmi := libvmi.New(
	libvmistatus.WithStatus(
		libvmistatus.New(libvmistatus.WithActivePod(types.UID(podUID), host)),
	),
)
```

sarthaksarthak9 commented on pkg/virt-handler/hotplug-disk/mount_test.go:
```diff
@@ -260,8 +260,10 @@ var _ = Describe("HotplugVolume", func() {
+			vmi = libvmi.New(
+				libvmi.WithName("fake-vmi"),
+				libvmi.WithUID("1234"),
+			)
```
@0xFelix, I tried implementing the active pods setting as you suggested, but it seems libvmi.WithActivePods() doesn’t exist in the current version of libvmi.

0xFelix replied:
As commented above, do it with `libvmistatus`

victortoso commented:
/cc
